GNUmakefile

```diff
@@ -1,36 +1,29 @@
-TEST?=$$(go list ./... |grep -v 'vendor')
-GOFMT_FILES?=$$(find . -name '*.go' |grep -v vendor)
+export GO111MODULE := on
+TEST?=$$(go list ./...)
+GOFMT_FILES?=$$(find . -name '*.go')
 
 default: build
 
-test: fmtcheck
+setup:
+    curl -sfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh| sh -s v1.21.0
+.PHONY: setup
+
+test:
 	@echo "==> Running tests..."
 	go test -i $(TEST) || exit 1
 	echo $(TEST) | \
 		xargs -t -n4 go test $(TESTARGS) -timeout=30s -parallel=4
-
-vet:
-	@echo "==> go vet"
-	go vet $$(go list ./... | grep -v vendor/) ; if [ $$? -eq 1 ]; then \
-		echo ""; \
-		echo "Vet found suspicious constructs. Please check the reported constructs"; \
-		echo "and fix them if necessary before submitting the code for review."; \
-		exit 1; \
-	fi
-
-fmt:
-	gofmt -s -w $(GOFMT_FILES)
-
-fmtcheck:
-	@sh -c "'$(CURDIR)/scripts/gofmtcheck.sh'"
-
-errcheck:
-	@sh -c "'$(CURDIR)/scripts/errcheck.sh'"
+.PHONY: test
 
 lint:
 	@echo "==> Linting all packages..."
 	golint -set_exit_status ./...
-	GOGC=30 golangci-lint run ./...
+	golangci-lint run ./... -E gofmt -E golint
```

Comment:
`golangci` already handles errcheck (enable by default) and can handle gofmt and golint if told to so no need for separate commands